wurst/objediting/UpgradeObjEditing.wurst

```diff
@@ -1,47 +1,530 @@
 package UpgradeObjEditing
 import ObjEditingNatives
 import NoWurst
+import public SharedEnums
+import Integer
+import Boolean
+import ErrorHandling
+
+constant BASE_UPGRADE_ID = 'Rhse' // magic sentry has no default effects and 1 level
+// Prefixes for the effects
+constant EFFECT_TAG_PREFIX = "gef"
+constant BASE_TAG_PREFIX = "gba"
+constant INCREMENT_TAG_PREFIX = "gmo"
+constant CODE_TAG_PREFIX = "gco"
+
+public enum UpgradeClass
+	NONE
+	Armor
+	Artillery
+	Melee
+	Ranged
+	Caster
+
+public enum UpgradeEffectType
+	NONE
```

Comment:
afaik the convention is all uppercase and underscore since these are basically constants.
In any case I wouldn't mix the two.